xds/src/main/java/io/grpc/xds/sds/SslContextSecretVolumeSecretProvider.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.envoyproxy.envoy.api.v2.auth.CertificateValidationContext;
+import io.envoyproxy.envoy.api.v2.auth.TlsCertificate;
+import io.envoyproxy.envoy.api.v2.core.DataSource.SpecifierCase;
+import io.grpc.netty.GrpcSslContexts;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+
+import java.io.File;
+import java.util.concurrent.Executor;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.SSLException;
+
+/**
+ * An SslContext provider that uses file-based secrets (secret volume).
+ * Used for both server and client SslContexts
+ */
+final class SslContextSecretVolumeSecretProvider
+    implements SecretProvider<SslContext> {
+  @VisibleForTesting
```

Comment:
I have removed as many of them as possible but the ones that are really used by test classes I like to keep them because of the testing coverage

Reply:
Removed `@VisibleForTesting` from the 2 factory methods. The others are needed because there is no other way to verify values like privateKey: they cannot be extracted from SslContext